Add plugin: Autofit Tabs

bwya77 opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/bwya77/autofit-tabs

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Zachatoo commented:
["Buy Me a Coffee": "buymeacoffee.com/bwya77",](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/manifest.json#L10)
Make sure to include the `https://` in your funding URLs.

Your main.css file isn't being referenced anywhere, please remove it from your repo if it isn't being used.

[import './styles.css';](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L2)
No need to import the CSS into your plugin, Obsidian will manage loading and unloading your styles for you if you include the styles.css file in your release.

[console.log('Autofit Tabs Plugin loading...');](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L33), [console.log('Unloading Autofit Tabs Plugin...');](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L46)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

[container.createEl('h2', { text: 'AutoFit Tabs Settings' });](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L247)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[container.createEl('h3', { text: title });](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L304)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

[this.createSettingsSection(container, 'Icon Settings', \[](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L267), [this.createSettingsSection(container, 'Other Settings', \[](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L280)
[Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)

[this.createSettingsSection(container, 'Basic Dimensions', \[](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L249), [name: 'Minimum Width',](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L251), [name: 'Close Button Width',](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L256), [name: 'Left Padding',](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L261), [name: 'Icon Width',](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L269), [name: 'Icon Width',](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L269), [name: 'Icon Right Margin',](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L274), [name: 'Active Indicator Width',](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L287), [name: 'Transition Duration',](https://github.com/bwya77/autofit-tabs/blob/6d1b7cc5e287552495b7a3008730115557ccf3df/main.ts#L292)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

bwya77 commented:
Made all changes in release 1.0.1

joethei commented:
["minAppVersion": "0.15.0",](https://github.com/bwya77/autofit-tabs/blob/8b87216b1924c57a897ab7fa69e79c01b915d99a/manifest.json#L5)
This is too low, tabs were only introduced in Obsidian 1.0.0

[this.createSettingsSection(container, 'Other', [](https://github.com/bwya77/autofit-tabs/blob/8b87216b1924c57a897ab7fa69e79c01b915d99a/main.ts#L274)
There should not be a header for "Generic" / "Other" settings.

bwya77 commented:
Fixed manifest minAppVersion to 1.0.0
removed the header for Other and placed at top without header [per guidelines](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)
made new release for changes
